### PR TITLE
Feature/139 actually run toolchain through gui

### PR DESCRIPTION
**Pull request by h-engelhardt**
- Now actually uses rules, concepts and mappings which were defined in the GUI
- If results cannot be computed an error message is shown
- For now it still shows the ouput view in case of an error for easier GUI development, but that should be changed at the end

### CODE REVIEW

File: web-ui/src/main/java/org/archcnl/domain/common/ArchitectureCheck.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+public class ArchitectureCheck {
+
+    private static final Logger LOG = LogManager.getLogger(ArchitectureCheck.class);
+
+    private ResultRepository repository;
+
+    private final String rootDir =
+            "D:/Programme/ArchCNL/ArchCNL/toolchain/src/integration-test/resources/";
+    private List<String> sourcePaths = Arrays.asList(rootDir + "OnionArchitectureDemo/src/");
```
Comment (AlexHamburg) [inline web-ui/src/main/java/org/archcnl/domain/common/ArchitectureCheck.java:26]: same

File: web-ui/src/main/java/org/archcnl/application/service/ConfigAppService.java [hunk 2]
```diff
@@ -49,6 +50,13 @@ public static String getDbPassword() throws PropertyNotFoundException {
                 ConfigAppService.DB_PASSWORD);
     }
 
+    /** @return DB context from property file */
+    public static String getDbContext() throws PropertyNotFoundException {
```
[the remark this answers is not in the document]
Comment (h-engelhardt) [reply web-ui/src/main/java/org/archcnl/application/service/ConfigAppService.java:54]: Test has been added. Has been added in docker-compose, but you should check if it is done how you meant it.

File: web-ui/src/main/java/org/archcnl/domain/common/ArchitectureCheck.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.archcnl.domain.output.repository.ResultRepositoryImpl;
+import org.archcnl.toolchain.CNLToolchain;
+
+public class ArchitectureCheck {
+
+    private static final Logger LOG = LogManager.getLogger(ArchitectureCheck.class);
+
+    private ResultRepository repository;
+
+    private final String rootDir =
```
[the remark this answers is not in the document]
Comment (h-engelhardt) [reply web-ui/src/main/java/org/archcnl/domain/common/ArchitectureCheck.java]: Has been removed as it actually was not necessary. Path is given when called.

File: web-ui/src/main/java/org/archcnl/ui/MainPresenter.java [hunk 4]
```diff
@@ -49,12 +59,57 @@ private void addListeners() {
...
+            architectureCheck = new ArchitectureCheck(path);
+            ResultRepository repository = architectureCheck.getRepository();
+            outputView.displayResult(
+                    repository.executeNativeSelectQuery(QueryUtils.getDefaultQuery()));
+            outputView.setResultRepository(repository);
+            view.showContent(outputView);
+        } catch (PropertyNotFoundException | StardogException | IOException e) {
+            final Notification notification = new Notification();
+            final Text errorMessage;
+            if (e instanceof PropertyNotFoundException) {
```
[the remark this answers is not in the document]
Comment (h-engelhardt) [reply web-ui/src/main/java/org/archcnl/ui/MainPresenter.java]: Simple switch case was not possible, as it does not work for instanceof and the exceptions can be subclasses of those.
Has been seperated into catch clauses for each exception type.

File: web-ui/src/main/java/org/archcnl/domain/common/ArchitectureCheck.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import org.archcnl.domain.output.repository.ResultRepositoryImpl;
+import org.archcnl.toolchain.CNLToolchain;
+
+public class ArchitectureCheck {
+
+    private static final Logger LOG = LogManager.getLogger(ArchitectureCheck.class);
+
+    private ResultRepository repository;
+    private List<String> sourcePaths;
+    private final String ruleFile = "temp/GeneratedRuleFile.adoc";
```
Comment (AlexHamburg) [inline web-ui/src/main/java/org/archcnl/domain/common/ArchitectureCheck.java:24]: pls as property

Comment (h-engelhardt) [reply web-ui/src/main/java/org/archcnl/domain/common/ArchitectureCheck.java]: done

**Pull request merged**